hnn_core/network.py

```diff
@@ -832,6 +903,320 @@ def _get_src_type_and_pos(self, gid):
+            List of string names of targets.
+        target_gids : list on int
+            List of integer identifiers of targets.
+        loc : str
+            If 'proximal' or 'distal', the corresponding
+            dendritic sections from Cell.sect_loc['proximal']
+            or Cell.Sect_loc['distal'] are used
+        receptor : str
+            The receptor.
+        nc_dict : dict
+            The connection dictionary containing keys
+            A_delay, A_weight, lamtha, and threshold.
+        allow_autapses : bool
+            If True, allow connecting neuron to itself.
+        unique : bool
+            If True, each target cell gets one "unique" feed.
+            If False, all src_type cells are connected to
+            all target_type cells.
+        """
+        connectivity = list()
+        src_start = src_gids[0]  # Necessary for unique feeds
+        for target_type, target_gid in zip(target_types, target_gids):
+            if unique:
+                src_gids = [target_gid + src_start]
+                src_types = [src_types[0]]  # Assumes that all src_types match
+            for src_type, src_gid in zip(src_types, src_gids):
+                conn = {'src_type': src_type, 'src_gid': src_gid,
+                        'target_type': target_type, 'target_gid': target_gid,
+                        'loc': loc, 'receptor': receptor,
+                        'nc_dict': deepcopy(nc_dict)}
+                if not allow_autapses and src_gid == target_gid:
+                    continue
+
+                connectivity.append(conn)
+        return connectivity
+
+    def add_connection(self, connectivity):
+        """Appends connections to connectivity list
+
+        Parameters
+        ----------
+        connectivity : dict | list of dict
+            Contains cell-cell connectivity information
+            in a dict, or list of dict with the following
+            items:
+
+            src_type : str
+                List of string names of sources.
+            src_gid : int
+                List of integer identifiers of sources.
+            target : str
+                List of string names of targets.
+            target_gid : int
+                List of integer identifiers of targets.
+            loc : str
+                If 'proximal' or 'distal', the corresponding
+                dendritic sections from Cell.sect_loc['proximal']
+                or Cell.Sect_loc['distal'] are used
+            receptor : str
+                The receptor.
+            nc_dict : dict
+                The connection dictionary containing keys
+                'A_delay', 'A_weight', 'lamtha', and 'threshold'.
+            allow_autapses : bool
+                If True, allow connecting neuron to itself.
+            unique : bool
+                If True, each target cell gets one "unique" feed.
+                If False, all src_type cells are connected to
+                all target_type cells.
```

Comment:
how is this relevant to the user?

Reply:
accidentally copied from old docstring, removed